[ja] Localize index page

ymotongpoo opened the pull request:
## Docs PR Checklist

<!--- Just making sure... -->

- [X] This PR is for a documentation page whose authoritative copy is in the
      opentelemetry.io repository.

svrnm commented:
Thanks @ymotongpoo, can you ask someone who is also a native Japanese speaker to provide a review? 

@chalin does this PR everything you need to add Japanese as a language to the website?

ymotongpoo commented:
@shirou @katzchang can either of you review the JA translation? The volume is pretty small.

shirou commented on content/ja/_index.md:
```diff
@@ -0,0 +1,95 @@
+</div>
+
+<div class="h3 mt-4">
+<a class="text-secondary" href="/docs/getting-started/">Get started</a> based on your role
```
Wouldn't it be better to translate this line too?

ymotongpoo replied:
thank you for catching this!

katzchang commented on content/ja/_index.md:
```diff
@@ -0,0 +1,95 @@
+
+{{% blocks/section color="secondary" type="cncf" %}}
+
+**OpenTelemetryは[CNCF][] [インキュベーティング][]プロジェクトです。**<br>OpenTracingとOpenCensusのプロジェクトが合併して設立されました。
```
incubatingは英単語のまま表記されるのが現在の慣例のようです
https://cloud-native.slack.com/archives/C057F81GFUG/p1713943821802049

ymotongpoo replied:
ack

katzchang commented on content/ja/_index.md:
```diff
@@ -0,0 +1,95 @@
+
+[cncf]: https://cncf.io
+[cncf logo]: /img/logos/cncf-white.svg
+[インキュベーティング]: https://www.cncf.io/projects/
```
https://github.com/open-telemetry/opentelemetry.io/pull/4580/files#r1625667262
と同様

ymotongpoo replied:
ack

chalin commented on hugo.yaml:
```diff
@@ -15,6 +15,10 @@ languages:
       description: The OpenTelemetry Project Site
+  ja:
+    languageName: 日本語 (Japanese)
+    # params:
+    #   description:
```
We'll need a Japanese translation of "The OpenTelemetry Project Site" to put in these lines. No rush, it can be handled in a followup PR.

chalin reviewed (approved):
While I still can't rebase, I have made all the necessary fixes. For a preview, see https://deploy-preview-4580--opentelemetry.netlify.app/ja/

IMHO, this is good to merge.

ymotongpoo commented:
@chalin Thank you for your review and I rebased and force pushed my branch to resolve conflicts. I also added metadata in Hugo config for Japanese pages.

svrnm commented:
https://opentelemetry.io/ja/ 🎉
